compiler/grainlsp/hacks.js

```diff
@@ -0,0 +1,57 @@
+//Provides: caml_ml_open_descriptor_in
+//Requires: caml_global_data,caml_sys_open,caml_raise_sys_error, caml_ml_channels
+//Requires: fs_node_supported, caml_string_of_jsstring
+function caml_ml_open_descriptor_in (fd)  {
+    var data = caml_global_data.fds[fd];
+    if(data.flags.wronly) caml_raise_sys_error("fd "+ fd + " is writeonly");
+    var refill = null;
+    if(fd == 0 && fs_node_supported()){
+      var fs = require('fs');
+      var Buffer = require('buffer').Buffer;
+      refill = function () {
+        var buf = Buffer.alloc(256);
+        var stdinFd = fs.openSync('/dev/stdin', 'rs');
+        var bytesRead = fs.readSync(fd, buf);
+        fs.closeSync(stdinFd);
+        // Ensures we don't have any trailing, empty data
+        var str = buf.slice(0, bytesRead).toString('utf8');
+        return caml_string_of_jsstring(str);
```

Comment:
cc @hhugo - would you be open to the changes in this file once I have everything working on Windows?